graph/Force-Directed-Graph/ForceDirectedGraph.tsx

```diff
@@ -0,0 +1,171 @@
+import React, { useEffect, useState } from "react";
+import { ActivityIndicator, View, Dimensions} from "react-native";
+import Svg, { Circle, Line} from "react-native-svg";
+
+import styles from "./styles";
+import Graph, { Link, Node } from "../Graph";
+import { fruchtermanReingold } from "../graph-algorithms/FruchtermanReingold";
+
+const DEFAULT_NODE_SIZE = 10; // Default size of the nodes
+const DEFAULT_NODE_SIZE_INCREMENT = 2; // Increment in the size of the nodes
+
+const DEFAULT_NODE_COLOR = "blue"; // Default color of the nodes
+const CONSTRAINED_NODE_COLOR = "red"; // Color of the constrained node
+
+const DEFAULT_LINK_COLOR = "black"; // Default color of the links
+
+const MAX_ITERATIONS = 1000; // Maximum number of iterations for the used algorithn
+
+const WIDTH = Dimensions.get("window").width; // Width of the screen
+const HEIGHT = Dimensions.get("window").height; // Height of the screen
+const CENTER_WIDTH = WIDTH / 2; // Center X-coordinates of the screen
+const CENTER_HEIGHT = HEIGHT / 2; // Center Y-coordinates of the screen
+
+/**
+ *
+ * @description Force-directed graph component
+ * @param graph - Graph object
+ * @param constrainedNodeId - Identifier of the node to be constrained to the center
+ * @returns - Force-directed graph component
+ */
+const ForceDirectedGraph: React.FC<{
+  graph: Graph;
+  constrainedNodeId: string;
+}> = ({ graph, constrainedNodeId }) => {
+  // States to store the nodes, links, sizes and loading status
+  const [nodes, setNodes] = useState<Node[]>([]);
+  const [links, setLinks] = useState<Link[]>([]);
+  const [sizes, setSizes] = useState<Map<string, number>>(new Map());
+  const [load, setLoad] = useState<boolean>(false);
+
+  // Use effect to update the graph
+  useEffect(() => {
+    // Get the initial links, nodes and sizes
+    const initialLinks = graph.getLinks();
+    const initialNodes = graph.getNodes();
+    const initialSizes = setNodesSizes([...initialLinks]);
+    // Set the links, nodes and sizes
+    setLinks([...initialLinks]);
+    setNodes(
+      fruchtermanReingold(
+        [...initialNodes],
+        initialLinks,
+        constrainedNodeId,
+        WIDTH,
+        HEIGHT,
+        MAX_ITERATIONS
+      )
+    );
+    setSizes(initialSizes);
+    setLoad(true);
+  }, [graph, constrainedNodeId]);
+
+  // If the graph is not loaded, display an activity indicator
+  if (!load) {
+    return <ActivityIndicator size="large" color="#0000ff" />;
+  }
```

Comment:
Could add the color in the dedicated file and import it so all colors are in the same place